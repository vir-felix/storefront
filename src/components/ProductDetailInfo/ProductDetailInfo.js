import React, { Component } from "react";
import PropTypes from "prop-types";
import Typography from "material-ui/Typography";
import Grid from "material-ui/Grid";

/**
 * Product detail basic info fields
 * @class ProductDetailTitle
 */
class ProductDetailInfo extends Component {
  static propTypes = {
    /**
     * Prodcut description
     */
    description: PropTypes.string,

    /**
     * Price or price range as a string
     */
    priceRange: PropTypes.string,

    /**
     * Product vendor
     */
    vendor: PropTypes.string
  }
  render() {
    const { priceRange, vendor, description } = this.props;

    // Render all props are undefined then skip rendering component
    if (!priceRange && !vendor && !description) return null;

    return (
      <Grid item sm={12}>
        {priceRange && <Typography component="div" gutterBottom={true} variant="title">{priceRange}</Typography>}
        {vendor && <Typography>{vendor}</Typography>}
        {description && <Typography>{description}</Typography>}
      </Grid>
    );
  }
}

export default ProductDetailInfo;
