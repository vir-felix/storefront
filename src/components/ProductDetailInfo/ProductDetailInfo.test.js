import React from "react";
import renderer from "react-test-renderer";
import ProductDetailInfo from "./ProductDetailInfo";

test("basic snapshot", () => {
  const component = renderer.create((
    <ProductDetailInfo
      priceRange="$19.99"
      vendor="Vendor"
      description="Description"
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot with price only", () => {
  const component = renderer.create((
    <ProductDetailInfo
      priceRange="$19.99"
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot with vendor only", () => {
  const component = renderer.create((
    <ProductDetailInfo
      vendor="Vendor"
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot with description only", () => {
  const component = renderer.create((
    <ProductDetailInfo
      description="Description"
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot with price and description", () => {
  const component = renderer.create((
    <ProductDetailInfo
      priceRange="$19.99"
      description="Description"
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot with no props", () => {
  const component = renderer.create(<ProductDetailInfo />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
