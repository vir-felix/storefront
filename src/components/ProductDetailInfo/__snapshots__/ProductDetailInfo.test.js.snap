// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-sm-12-51"
>
  <div
    className="MuiTypography-root-91 MuiTypography-title-97 MuiTypography-gutterBottom-108"
  >
    $19.99
  </div>
  <p
    className="MuiTypography-root-91 MuiTypography-body1-100"
  >
    Vendor
  </p>
  <p
    className="MuiTypography-root-91 MuiTypography-body1-100"
  >
    Description
  </p>
</div>
`;

exports[`snapshot with description only 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-sm-12-51"
>
  <p
    className="MuiTypography-root-91 MuiTypography-body1-100"
  >
    Description
  </p>
</div>
`;

exports[`snapshot with no props 1`] = `null`;

exports[`snapshot with price and description 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-sm-12-51"
>
  <div
    className="MuiTypography-root-91 MuiTypography-title-97 MuiTypography-gutterBottom-108"
  >
    $19.99
  </div>
  <p
    className="MuiTypography-root-91 MuiTypography-body1-100"
  >
    Description
  </p>
</div>
`;

exports[`snapshot with price only 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-sm-12-51"
>
  <div
    className="MuiTypography-root-91 MuiTypography-title-97 MuiTypography-gutterBottom-108"
  >
    $19.99
  </div>
</div>
`;

exports[`snapshot with vendor only 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-sm-12-51"
>
  <p
    className="MuiTypography-root-91 MuiTypography-body1-100"
  >
    Vendor
  </p>
</div>
`;
